spring-ai-core/src/main/java/org/springframework/ai/client/metadata/RateLimit.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.client.metadata;
+
+import java.time.Duration;
+
+/**
+ * Abstract Data Type (ADT) encapsulating metadata from an AI provider's API rate limits
+ * granted to the API key in use and the API key's current balance.
+ *
+ * @author John Blum
+ * @since 0.7.0
+ */
+public interface RateLimit {
```

Comment:
similar to the comment on Usage, I'd like to compare what is out there, a bit harder to find this information from a quick google search as compared to usage.

Reply:
ACK